### DISCUSSION TITLE
skywalking lost services registered

### DISCUSSION THREAD

**Discussion by baowp** [apache/skywalking#12549] [Q&A] [answered]
We have connected 50+ java microservices to skywalking-oap cluster with 2nodes and keep on adding more into it. But later we find some of the services are not available on skywalking-ui, and also find some services available on skywalking-ui have no instances. What may cause the problems.  The segment data collected from agents is around 5000/s and metrcs data around 1000/s according to the writing speed from ES. Can some people help with the issues. We use skywalking 9.7.0 and agent 9.1.0

**Comment by wu-sheng** [comment]
You keep adding but don't scale, it must fail and begin to drop data for sure.

What do expect us to say?

**Reply by baowp** [reply]
If some data are dropped, will it only affect particular services? If parts of the metrics are lost in the charts, we can assume that the data has been dropped. The current situation involves several services not showing up on the SW-UI, and several services have intact charts but no instances under their instance tab. The alarm messages still include the missing services.

**Reply by wu-sheng** [reply]
Service list is also a metric.

**Reply by wu-sheng** [reply]
Any metric could be dropped without warning.

**Reply by baowp** [reply]
Is the loss  of the service list metrics data so fixed, and the unavailable services always unseen? 
We attempted to add a new OAP node to the cluster and configured one of the missing services to report data specifically to this new OAP. However, we still cannot see the service displayed on the SW-UI.

**Reply by wu-sheng** [reply]
> Is the loss of the service list metrics data so fixed, and the unavailable services always unseen?

What do you mean?

**Reply by wu-sheng** [reply]
Hve you checked the service reporting telemetry data?
I feel it isn't.

**Reply by baowp** [reply]
> Hve you checked the service reporting telemetry data?

We configured those missing services as the same way as other services, and see the same agent log output

**Reply by baowp** [reply]
> > Is the loss of the service list metrics data so fixed, and the unavailable services always unseen?
> 
> What do you mean?

I mean the missing services are always those services

**Reply by wu-sheng** [reply]
I don't think that is the data drop, it should be something wrong.
Agent config, OAP log failing, or something else.

**Reply by baowp** [reply]
We configured every service and instance in the same way. Initially, all services were visible, but later they became unseen on the SW-UI. We did not modify the agent configuration. How would an agent fail when nothing changed

**Reply by wu-sheng** [reply]
I don't know that. 

You should check, you keep saying everything should be good, this doesn't help.

**Reply by baowp** [reply]
Do the OAP logs indicate what went wrong, or do agent logs? We couldn't get useful information from these logs

**Reply by wu-sheng** [reply]
I could only say, that if there is a network issue, the logs would print. But I am not sure what is your issue.

**Reply by baowp** [reply, marked as answer]
We set up a dedicated es cluster for storing skywalking data and upgraded es7 to es8. The previous es cluster is used for storing logs collected by filebeat from a lot of microservices. With the new dedicated es cluster, all services and instances are now available on the UI
